src/anet.c

```
@@ -381,16 +381,16 @@ static int anetSetReuseAddr(char *err, int fd) {
     return ANET_OK;
 }
 
-static int anetCreateSocket(char *err, int domain) {
+static int anetCreateSocket(char *err, int domain, int type, int protocol, int sockopts) {
     int s;
-    if ((s = socket(domain, SOCK_STREAM, 0)) == -1) {
+    if ((s = socket(domain, type, protocol)) == ANET_ERR) {
         anetSetError(err, "creating socket: %s", strerror(errno));
         return ANET_ERR;
     }
 
     /* Make sure connection-intensive things like the redis benchmark
      * will be able to close/open sockets a zillion of times */
-    if (anetSetReuseAddr(err,s) == ANET_ERR) {
+    if (sockopts & SO_REUSEADDR && anetSetReuseAddr(err,s) == ANET_ERR) {
```

i see you didn't pass this flag in anetUnixServer, shall i assume it's not relevant in this case and so far it failed silently?

---

`SO_REUSEADDR` here seems more like a specific usage, I think this won't affect the general tests. But I'll add it back